django/db/models/sql/query.py

```diff
@@ -441,17 +441,28 @@ def get_aggregation(self, using, added_aggregate_names):
         """
         if not self.annotation_select:
             return {}
-        existing_annotations = [
-            annotation
-            for alias, annotation in self.annotations.items()
+        existing_annotations = {
+            alias: annotation
+            for alias, annotation in self.annotation_select.items()
             if alias not in added_aggregate_names
-        ]
+        }
+        existing_aggregations = {
+            alias
+            for alias, annotation in existing_annotations.items()
+            if getattr(annotation, "contains_aggregate", True)
+            or getattr(annotation, "contains_over_clause", True)
+        }
+        # Some backends don't support the usage of HAVING without an explicit
+        # GROUP BY (e.g. SQLite).
+        requires_grouping = not connections[
+            using
+        ].features.supports_having_without_group_by and any(
+            self.where.split_having_qualify()[1:]
+        )
```

Comment:
SQLite errors with `django.db.utils.OperationalError: a GROUP BY clause is required before HAVING` without this branch.